Tests/AsyncQueueTests/ActorQueueTests.swift

```diff
@@ -31,54 +31,99 @@ final class ActorQueueTests: XCTestCase {
     override func setUp() async throws {
         try await super.setUp()
 
-        systemUnderTest = ActorQueue()
+        systemUnderTest = ActorQueue<Counter>()
+        counter = Counter()
+        systemUnderTest.adoptExecutionContext(of: counter)
     }
 
     // MARK: Behavior Tests
 
+    func test_adoptExecutionContext_doesNotRetainActor() {
+        let systemUnderTest = ActorQueue<Counter>()
+        var counter: Counter? = Counter()
+        weak var weakCounter = counter
+        systemUnderTest.adoptExecutionContext(of: counter!)
+        counter = nil
+        XCTAssertNil(weakCounter)
+    }
+
+    func test_async_retainsAdoptedActorUntilQueueFlushes() async {
```

Comment:
This test doesn't flush the queue. Can we rename it to something about how the actor is retained while there is in-progress tasks?

Reply:
Technically we are flushing the queue by waiting for the `async` task to complete, but that's not super intuitive! Updated in 1c8cab5